src/casync-tool.c

```diff
@@ -1629,6 +1662,14 @@ static int pull(int argc, char *argv[]) {
                 return r;
         }
 
+        if (arg_rate_limit > 0) {
+                r = ca_remote_set_rate_limit(rr, arg_rate_limit);
+                if (r < 0) {
+                        fprintf(stderr, "Failed to set rate limit: %s\n", strerror(-r));
+                        return r;
+                }
+        }
```

Comment:
maybe turn this 7 lines above into a new helper load_rate_limit() or so, much like load_feature_flags() and load_chunk_size(), which we can then reuse for each method